packages/telemetry/README.md

```diff
@@ -0,0 +1,278 @@
+deactivate Cypress Server
+activate Cypress App
+Cypress App->>Cypress App: Initialize  Browser Telemetry Singleton
+deactivate Cypress App
+par Browser Span lifecycle
+Cypress App->>Cypress App: Span Starts
+activate Cypress App
+Cypress App->>Cypress Server: Span Ends, <br>send telemetry to Cypress server.
+deactivate Cypress App
+activate Cypress Server
+Note right of Cypress App: Uses the web socket to avoid <br> network calls that would show up<br> in Cypress logs
+and Child Process Span Lifecycle
+Cypress Child Process->>Cypress Child Process: Span Starts
+activate Cypress Child Process
+Cypress Child Process->>Cypress Server: Span Ends, <br>send telemetry to Cypress server.
+deactivate Cypress Child Process
+Note over Cypress Child Process, Cypress App: Uses the IPC to avoid <br> encrypting in the child process
+end
+Cypress Server->>Cypress Cloud: Encrypt Span, forward to Cloud
+deactivate Cypress Server
+activate Cypress Cloud
+Note right of Cypress Server: Telemetry data will always be <br>encrypted
+alt Cypress project
+Cypress Cloud->>Honeycomb: Decrypt span data, forward to honeycomb
+else not Cypress project
+Cypress Cloud->>Cypress OTEL Store: Decrypt span data, forward to OTEL Store(future)
+deactivate Cypress Cloud
+Note right of Cypress Cloud: For now we could dead end this data <br> until we're ready to store more.
+end
+```
+
+## Setup
+
+To prepare the telemetry singleton for use you first need to initialize it. This should be done at the start of whatever process you wish to monitor.
+
+There are two different singletons included in the telemetry package, one for node and one for browser. They have different requirements to set up.
+
+### Node
+
+To access the node telemetry singleton use the default export.
+
+```js
+const { telemetry } = require('@packages/telemetry')
+
+telemetry.init({options})
+```
+
+The node telemetry instance has a couple of exporters to choose from, the `OTLPTraceExporterCloud` and `OTLPTraceExporterIpc`.
+
+#### OTLPTraceExporterCloud
+
+The `OTLPTraceExporterCloud` will send telemetry as an http request to the supplied url and can be configured to encrypt the data. Within Cypress the exporter will encrypt data by default and send it to the cloud telemetry endpoint.
+
+```js
+const { OTLPTraceExporterCloud } = require('@packages/telemetry')
+
+const exporter = new OTLPTraceExporterCloud({
+  url: apiRoutes.telemetry(),
+  encryption,
+})
+```
+
+When sending data to the cloud telemetry endpoint you must attach the project id to the exporter as a header when it is available.
+
+```js
+const { telemetry, OTLPTraceExporterCloud } = require('@packages/telemetry')
+
+(telemetry.exporter() as OTLPTraceExporterCloud)?.attachProjectId(config.projectId)
+
+```
+
+When developing Cypress locally it is possible to override the telemetry endpoint and send unencrypted data to your own honeycomb instance. This is recommended when verifying newly added spans and any sort of local telemetry development.
```

Comment:
should we link to this in `start-cypress`?

Reply:
done